httpClient/src/main/java/org/jfrog/artifactory/client/httpClient/http/auth/PreemptiveAuthInterceptor.java

```diff
@@ -20,20 +24,51 @@
  */
 public class PreemptiveAuthInterceptor implements HttpRequestInterceptor {
 
+    private static final String ORIGINAL_HOST_CONTEXT_PARAM = "original.host.context.param";
+
     @Override
     public void process(HttpRequest request, HttpContext context) throws HttpException, IOException {
+
+        if (!shouldSetAuthScheme(request, context)) {
+            return;
+        }
+
         HttpClientContext clientContext = HttpClientContext.adapt(context);
         AuthState authState = clientContext.getTargetAuthState();
-
         // If there's no auth scheme available yet, try to initialize it preemptively
         if (authState.getAuthScheme() == null) {
-            CredentialsProvider credsProvider = clientContext.getCredentialsProvider();
-            HttpHost targetHost = clientContext.getTargetHost();
-            Credentials creds = credsProvider.getCredentials(
-                    new AuthScope(targetHost.getHostName(), targetHost.getPort()));
-            if (creds != null) {
-                authState.update(new BasicScheme(), creds);
+            String accessToken = clientContext.getUserToken(String.class);
+            if (StringUtils.isNotEmpty(accessToken)) {
+                request.addHeader("Authorization", "Bearer " + accessToken);
+            } else {
+                CredentialsProvider credsProvider = clientContext.getCredentialsProvider();
+                HttpHost targetHost = clientContext.getTargetHost();
+                Credentials creds = credsProvider.getCredentials(
+                        new AuthScope(targetHost.getHostName(), targetHost.getPort()));
+                if (creds == null) {
+                    throw new HttpException("No credentials for preemptive authentication");
+                }
+                BasicScheme authScheme = new BasicScheme();
+                authState.update(authScheme, creds);
             }
         }
     }
+
+    private boolean shouldSetAuthScheme(final HttpRequest request, final HttpContext context) {
```

Comment:
The code of this method is not indented correctly.

Reply:
Fixed